src/cdk-experimental/column-resize/resize-strategy.ts

```diff
@@ -0,0 +1,185 @@
+ */
+
+import {Inject, Injectable, OnDestroy, Provider} from '@angular/core';
+import {DOCUMENT} from '@angular/common';
+import {coerceCssPixelValue} from '@angular/cdk/coercion';
+
+import {ColumnResize} from './column-resize';
+
+/**
+ * Provides an implementation for resizing a column.
+ * The details of how resizing works for tables for flex mat-tables are quite different.
+ */
+@Injectable()
+export abstract class ResizeStrategy {
+  abstract applyColumnSize(
+      cssFriendlyColumnName: string,
+      columnHeader: HTMLElement,
+      sizeInPx: number): void;
+
+  abstract applyMinColumnSize(
+      cssFriendlyColumnName: string,
+      columnHeader: HTMLElement,
+      minSizeInPx: number): void;
+
+  abstract applyMaxColumnSize(
+      cssFriendlyColumnName: string,
+      columnHeader: HTMLElement,
+      minSizeInPx: number): void;
+}
+
+/**
+ * The optimially performing resize strategy for &lt;table&gt; elements with table-layout: fixed.
+ * Tested against and outperformed:
+ *   CSS selector
+ *   CSS selector w/ CSS variable
+ *   Updating all cell nodes
+ */
+@Injectable()
+export class TableLayoutFixedResizeStrategy extends ResizeStrategy {
+  applyColumnSize(_: string, columnHeader: HTMLElement, sizeInPx: number): void {
+    columnHeader.style.width = coerceCssPixelValue(sizeInPx);
+  }
+
+  applyMinColumnSize(_: string, columnHeader: HTMLElement, sizeInPx: number): void {
+    columnHeader.style.minWidth = coerceCssPixelValue(sizeInPx);
+  }
+
+  applyMaxColumnSize(): void {
+    // Intentionally omitted as max-width causes strange rendering issues in Chrome.
+    // Max size will still apply when the user is resizing this column.
+  }
+}
+
+/**
+ * The optimally performing resize strategy for flex mat-tables.
+ * Tested against and outperformed:
+ *   CSS selector w/ CSS variable
+ *   Updating all mat-cell nodes
+ */
+@Injectable()
+export class CdkFlexTableResizeStrategy extends ResizeStrategy implements OnDestroy {
+  private readonly _document: Document;
+  private readonly _columnIndexes = new Map<string, number>();
+  private readonly _columnProperties = new Map<string, Map<string, string>>();
+
+  private _styleElement?: HTMLStyleElement;
+  private _indexSequence = 0;
+
+  protected readonly defaultMinSize = 0;
+  protected readonly defaultMaxSize = Number.MAX_SAFE_INTEGER;
+
+  constructor(
+      private readonly _columnResize: ColumnResize,
+      @Inject(DOCUMENT) document: any) {
+    super();
+    this._document = document;
+  }
+
+  applyColumnSize(cssFriendlyColumnName: string, _: HTMLElement, sizeInPx: number): void {
+    const cssSize = coerceCssPixelValue(sizeInPx);
+
+    this._applyProperty(cssFriendlyColumnName, 'flex', `0 0.01 ${cssSize}`);
+  }
+
+  applyMinColumnSize(cssFriendlyColumnName: string, _: HTMLElement, sizeInPx: number): void {
+    const cssSize = coerceCssPixelValue(sizeInPx);
+
+    this._applyProperty(cssFriendlyColumnName, 'min-width', cssSize,
```

Comment:
Have you tried this against IE/Edge? They tend to not work very well with `min-width`/`min-height` and flexbox.

Reply:
Honestly I'm not even sure how I'd test that (I only have Macs and Linux boxes here at work). Is this still a concern with Chromium-based Edge?

Reply:
Probably not, but the Chromium-based Edge is still away from being the default.